src/ImageSharp/Formats/Heif/Av1/Transform/Forward/Av1Dct4ForwardTransformer.cs

```diff
@@ -35,51 +68,71 @@ public static void TransformSse(ref Vector128<int> input, ref Vector128<int> out
         Vector128<int> v0, v1, v2, v3;
 
         int endidx = 3 * columnNumber;
-        s0 = Sse41.Add(input, Unsafe.Add(ref input, endidx));
-        s3 = Sse41.Subtract(input, Unsafe.Add(ref input, endidx));
+        s0 = Sse2.Add(input, Unsafe.Add(ref input, endidx));
```

Comment:
If possible, we should look to use `Vector128` opreators and methods here that should abstract the instrinsic support.

I'm a little nervous also about the lack of contraints here. Are they strictly defined in the caller?

Reply:
The constraints are handled inside the `Av1BlockDecoder.DeriveBlockPointers` method. Please note that the `Av1FrameBuffer` is over allocating, such that the last blocks still fit in entirely (even if the resulting image is smaller). 

Please note, the `Transform` method will be called twice for every 16 pixels, for each color channel. In other words, this is bound to become a performance hotspot !

Reply:
The operators and methods are hardware accelerated though and provide additional support for AdvSimd and PackedSimd